packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -726,6 +722,16 @@ export class AppSyncModelVisitor<
         return true;
       });
     });
+
+    Object.values(this.modelMap).forEach(model => {
+      model.fields.forEach(field => {
+        const connectionInfo = field.connectionInfo;
+        if (connectionInfo?.kind === CodeGenConnectionType.BELONGS_TO && connectionInfo.targetName !== 'id') {
```

Comment:
I'd consider making this a negative case, since you probably wantit for any case where the kind is NOT HAS_MANY or HAS_ONE based on the logic above.